### DISCUSSION TITLE
Bubble Multiplier in MINI element

### DISCUSSION THREAD

**Discussion by adtzlr** [kinnala/scikit-fem#1069] [Q&A] [answered]
Hi,

would you be so kind to guide me to a literature reference which describes from where the value of the bubble multiplier (256) for the MINI element comes from?

https://github.com/kinnala/scikit-fem/blob/29ac66cac6c60cbf9ed328d8976794ed1b343f49/skfem/element/element_tet/element_tet_mini.py#L37

Thanks!

### Edit
I'll have a look here: https://github.com/kinnala/scikit-fem/discussions/722#discussioncomment-1253247. Thanks!

**Comment by gdmcbain** [comment, marked as answer]
I have actually lost access to my original notes (they being locked up at a previous employer's) but looking at the [blame](https://github.com/kinnala/scikit-fem/blame/29ac66cac6c60cbf9ed328d8976794ed1b343f49/skfem/element/element_tet/element_tet_mini.py#L37) for  `ElementTetMini` reveals quite an informative commit message for that line: [choose 256 as bubble coefficient to take unity at centroid](<https://github.com/kinnala/scikit-fem/commit/026551cbaa571d2b848393b2165e6d747f62bdcb>).  From memory, I just worked that out by hand.

**Reply by gdmcbain** [reply]
I don't think the value matters much, the 256 just changes the value of the degree of freedom inversely.  The bubble function isn't a Lagrangian interpolant.

**Reply by kinnala** [reply]
From stability viewpoint, it does not matter what value is picked there and you could, in fact, pick _any bubble function_ which is zero on the boundary of every element. This I have proven many times while working as a teaching assistant on a finite element stability course. In Stokes, stability is about the uniqueness of the pressure field and this is what you get for any coefficient.

From error estimate (i.e. uniform stability) viewpoint, I'm guessing it will scale the inf-sup constant (perhaps inversely?) and, therefore, be part of the unknown coefficient in the a priori error estimate. So it does have some effect in the error but asymptotically I think all of them are equivalent. This I have not proven myself and is just a gut instinct.

**Reply by adtzlr** [reply]
Sorry for my delayed response and thank you both @kinnala and @gdmcbain for your immediate answers/comments! I'll definitely have to pin error estimates and stability on top of my fea to-do list. In the meantime, I have the feeling that a bubble multiplier of one, which turned out to work well for nearly-incompressible hyperelasticity, is not entirely wrong.

**Reply by kinnala** [reply]
Yes. It does not matter which coefficient is used.
